blocks/button/edit.tsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect } from 'react';
+import { __ } from '@wordpress/i18n';
+import { useBlockProps, InspectorControls, InnerBlocks } from '@wordpress/block-editor';
+import { ColorPicker, PanelBody } from '@wordpress/components';
+import useInnerBlockAttributes from '@/hooks/useInnerBlocksAttributes';
+
+/**
+ * Lets webpack process CSS, SASS or SCSS files referenced in JavaScript files.
+ * Those files can contain any CSS code that gets applied to the editor.
+ *
+ * @see https://www.npmjs.com/package/@wordpress/scripts#using-css
+ */
+// Uncomment this line if you want to import a CSS file for this block.
+// import './index.scss';
+
+/**
+ * The edit function describes the structure of your block in the context of the
+ * editor. This represents what the editor will render when the block is used.
+ *
+ * @see https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#edit
+ *
+ * @return {WPElement} Element to render.
+ */
+interface EditProps {
+  attributes: {
+    bgColor?: string;
+    btnWidth?: string,
+    radius?: string,
+    textColor?: string,
+  };
+  setAttributes: (attributes: {}) => void;
+  clientId: string,
+}
+
+export default function Edit({
+  attributes: {
+    bgColor = '#0279af',
+    btnWidth = 'max-content',
+    radius = '0',
+    textColor = '#fff',
+  },
+  setAttributes,
+  clientId,
+}: EditProps) {
+  const TEMPLATE = [['core/button']];
+  const innerBlockAttributes = useInnerBlockAttributes(clientId);
+  const innerBorderRadius = innerBlockAttributes[0]?.style?.border?.radius || '0';
+  const innerWidth = innerBlockAttributes[0]?.width;
+  const buttonStyles = {
+    backgroundColor: bgColor,
+    borderRadius: radius,
+    color: textColor,
+    margin: '0 auto',
+    width: btnWidth,
+  };
+
+  useEffect(() => {
+    setAttributes({
+      radius: innerBorderRadius,
+      btnWidth: innerWidth !== undefined ? `${innerWidth}%` : 'max-content',
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Is disabling this ESLint rule still needed? It looks like all dependencies are listed. Could we remove?

Reply:
eslint wants `setAttributes` to be a part of the dependency array, so we still need this unfortunately